components/remote_settings/src/storage.rs

```diff
@@ -82,21 +84,33 @@ impl Storage {
     ) -> Result<Option<Vec<RemoteSettingsRecord>>> {
         let tx = self.conn.transaction()?;
 
-        // Check if the collection has been fetched before
+        let jexl_context = create_jexl_context(&self.context);
+
         let fetched: Option<bool> = tx
             .prepare("SELECT fetched FROM collection_metadata WHERE collection_url = ?")?
             .query_row(params![collection_url], |row| row.get(0))
             .optional()?;
 
         let result = match fetched {
             Some(true) => {
-                // If fetched before, get the records from the records table
-                let records: Vec<RemoteSettingsRecord> = tx
-                    .prepare("SELECT data FROM records WHERE collection_url = ?")?
-                    .query_map(params![collection_url], |row| row.get::<_, Vec<u8>>(0))?
-                    .map(|data| serde_json::from_slice(&data.unwrap()).unwrap())
-                    .collect();
-
+                let mut stmt = tx.prepare("SELECT data FROM records WHERE collection_url = ?")?;
+                let mut rows = stmt.query(params![collection_url])?;
+
+                let mut records = Vec::new();
+                while let Some(row) = rows.next()? {
+                    let data: Vec<u8> = row.get(0)?;
+                    let record: RemoteSettingsRecord = serde_json::from_slice(&data)?;
+
+                    if let Some(filter_expr) = &record.filter_expression {
+                        let eval_result = evaluate_filter_expression(filter_expr, &jexl_context);
```

Comment:
Are we sure we want to execute the filtering at the storage level? 
That means every type of storage will have to duplicate this code. Why not do it at a higher level?

Reply:
Yeah that's an interesting question. I don't see that we have different types of storage, parallel to each other. I decided to keep the `client` as clean and unchanged as possible, and we might want to store the `context` in the database as well? But maybe not. 

But I agree also with the other points: Nimubs has already implemented the filters. We could create a shared `jexl` crate and share it between components?

@jeddai what do you think? We could move the `jexl` code from Nimbus out into it's own crate, and share it across components?

Reply:
> we might want to store the context in the database as well? But maybe not.

I don't think so. Those are like build-time constants

Reply:
Moved it out!